parcels/particleset.py

```diff
@@ -528,6 +528,10 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
             if abs(time-next_movie) < tol:
                 self.show(field=movie_background_field, show_time=time, animation=True)
                 next_movie += moviedt * np.sign(dt)
+            # ==== insert post-process here to also allow for memory clean-up via external func ==== #
+            if postIterationFunctions is not None:
+                for extFunc in postIterationFunctions:
+                    extFunc()
```

Comment:
This is a new feature that's now buried in this larger PR; would it make sense to separate it out? And then also add a unit test so that we test its working?

More fundamentally, I'm not sure this feature is even necessary. In principle, a user could do the same with a for loop around the `pset.execute()` and setting `runtime` and `dt` the same value. Then, the execute would only do one kernel call each time. Admittedly, this adds a bit of overhead on the start of `pset.execute()`, but that's probably fairly limited.

So aren't we over-engineering by adding this feature?

Reply:
a) put it in there cause otherwise the whole performance measuring script doesn't work, as we can't obtain performance metrics otherwise at each iteration
b) yes, agreed - it may be a bit of its own PR, but again: I need that function for the example script anyway.
c) it's not useless - see #723 
d) the feature is necessary, cause in the for loop, you also then measure the setup time for the ParticleSet.__init__ (considerable), and the first-kernel start-up including the JIT compilation of the kernels - in essence: the measurements are deeply flawed, at would not survive CS paper peer review. This way, the first measurement taken (the blue spike at the start of the graphs) includes the whole setup procedure (JIT compilation etc.), and the other measurements are really the per-iteration measurements without overhead.